ansible_builder/main.py

```diff
@@ -145,62 +150,68 @@ def get_dependency(self, entry):
 
         return os.path.join(self.reference_path, req_file)
 
-    def collection_dependencies(self, dep_type='python'):
-        """Returns a list of files for the dependency type
-        These are the dependency files declared by collections which
-        the user definition listed in its Galaxy requirement file
-        in other words, this returns indirect dependencies of given type
-        """
-        ret = []
-        for path in self.manager.path_list():
-            CD = CollectionDefinition(path)
-            dep_file = CD.get_dependency(dep_type)
-            if not dep_file:
-                continue
-            namespace, name = CD.namespace_name()
-            ret.append(os.path.join(namespace, name, dep_file))
-        return ret
-
 
 class Containerfile:
     newline_char = '\n'
 
     def __init__(self, definition,
                  filename=constants.default_file,
                  build_context=constants.default_build_context,
-                 base_image=constants.default_base_image):
+                 base_image=constants.default_base_image,
+                 container_runtime=constants.default_container_runtime,
+                 tag=constants.default_tag):
 
         self.build_context = build_context
         os.makedirs(self.build_context, exist_ok=True)
         self.definition = definition
         self.path = os.path.join(self.build_context, filename)
         self.base_image = base_image
+        self.container_runtime = container_runtime
+        self.tag = tag
+        self.steps = []
```

Comment:
The code `self.steps =` shouldn't occur twice. This should be replaced by

```python
        self.steps = [
            "FROM {}".format(self.base_image),
            ""
        ]
```

and have that removed from `prepare_galaxy_steps`. As it is it both prepares galaxy steps and initializes the steps... although they have already been initialized.